dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/AbstractJdbcEventAnalyticsManager.java

```diff
@@ -417,12 +418,36 @@ else if ( queryItem.getValueType() == ValueType.NUMBER && !isGroupByClause )
                 columnAndAlias.getColumn(),
                 defaultIfNull( columnAndAlias.getAlias(), queryItem.getItemName() ) );
         }
+        else if ( queryItem.isText() && !isGroupByClause && hasOrderByClauseQueryItem( queryItem, params ) )
+        {
+            return getColumnAndAliasWithNullIfFunction( queryItem );
+        }
         else
         {
             return getColumnAndAlias( queryItem, isGroupByClause, "" );
         }
     }
 
+    private ColumnAndAlias getColumnAndAliasWithNullIfFunction( QueryItem queryItem )
```

Comment:
Would be nice to document(Javadoc) this method and also give an example of the output expected.